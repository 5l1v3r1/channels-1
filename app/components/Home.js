// @flow
import React, { Component } from 'react';
import LeftColumn from '../containers/LeftColumn';
import RightColumn from '../containers/RightColumn';
import Loader from './core/Loader';
import ErrorState from './core/ErrorState';
import styles from './Home.scss';

export default class Home extends Component {
  props: {
    token: string,
    isFetching: boolean,
    error: boolean,
    fetchAccessToken: () => Promise<any>
  };

  intervalId: number;
  fiftyMinutes: number;
  interval: () => number;

  constructor(props: any) {
    super(props);
    this.fiftyMinutes = 3000000;
    this.interval = () => setInterval(this.props.fetchAccessToken, this.fiftyMinutes);
  }

  componentDidMount() {
    this.props.fetchAccessToken();
    this.intervalId = this.interval();
  }

  componentWillUnmount() {
    clearInterval(this.intervalId);
  }

  render() {
    const {
      token,
      isFetching,
      error,
      fetchAccessToken
    } = this.props;
    if (!token || error) {
      return (
        <div className={styles.home}>
          {isFetching ? (
            <Loader />
          ) : (
            <ErrorState
              message={'Error accessing your account.'}
              retry={fetchAccessToken}
            />
          )}
        </div>
      );
    }

    return (
      <div className={styles.home}>
        <LeftColumn />
        <RightColumn />
      </div>
    );
  }
}
