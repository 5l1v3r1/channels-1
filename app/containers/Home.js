import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as accessTokenActions from '../actions/accessToken';
import Home from '../components/Home';
import { getActiveChannel } from '../reducers/index';

function mapStateToProps(state) {
  return {
    token: state.accessToken.token,
    isFetching: state.accessToken.isFetching,
    error: state.accessToken.error,
    activeChannel: getActiveChannel(state),
    activeVideo: state.activeVideo
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(accessTokenActions, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);
