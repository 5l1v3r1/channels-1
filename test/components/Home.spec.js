import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from '../../app/components/Home';

jest.mock('../../app/containers/LeftColumn', () => 'LeftColumn');
jest.mock('../../app/containers/RightColumn', () => 'RightColumn');

function setup() {
  const props = {
    token: '',
    isFetching: false,
    error: false,
    fetchAccessToken: jest.fn()
  };

  const enzymeWrapper = shallow(<Home {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Home', () => {
  it('should render self with loader', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      isFetching: true
    });
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should render self with error message', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      error: true
    });
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should render self with content', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      token: 'ya29.GlyrBCJQJoIYFzocIunVN-CfjQZMG4oyVuAB6v_x_Z3FRnViyPy_deqRdwSAipQtKc9Nb2RudM9UISwI8SGNXxsJ1t3QHddeCdnoCjsM_vhLa9FlFVqMN_seI7oljg'
    });
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should call fetchAccessToken and setInterval on componentDidMount', () => {
    const spy = jest.spyOn(Home.prototype, 'componentDidMount');
    const { props } = setup();
    const enzymeWrapper = mount(<Home {...props} />);

    expect(spy).toBeCalled();
    expect(enzymeWrapper.instance().intervalId).toBeDefined();
  });
});
