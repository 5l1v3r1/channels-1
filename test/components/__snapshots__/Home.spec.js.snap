// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render self with content 1`] = `
<div
  className="home"
>
  <LeftColumn />
  <RightColumn />
</div>
`;

exports[`Home should render self with error message 1`] = `
<div
  className="home"
>
  <ErrorState
    color="black"
    message="Error accessing your account."
    retry={[Function]}
  />
</div>
`;

exports[`Home should render self with loader 1`] = `
<div
  className="home"
>
  <Loader />
</div>
`;
